server/schemas/item.graphql

```diff
@@ -71,6 +71,13 @@ extend type Query {
     before: Cursor
   ): ItemConnection!
   versionsByItem(itemId: ID!): [VersionedItem!]!
+  searchItem(
+    key: String!
```

Comment:
Given that we will extend more advanced queries in the future, the current input is rough. Let's define the query in a bit more detail.

```gql
input ItemQuery {
  workspace: ID!
  project: ID!
  q: String
  # add more queries here in the future
}
```

Then

```suggestion
    key: ItemQuery!
```

Also at `pkg/item`, define domain model:

```go
type Query struct {
  workspace id.WorkspaceID
  project id.ProjectID
  q string
}

func NewQuery(workpsace id.ProjectID, project *id.ProjectID, q string) *Query {
	return &Query{
		workpsace: workspace,
		project: project,
		q: q,
	}
}

// Q returns keywords for search
func (q *Query) Q() string {
	return q.q
}

// define Workspace and Project method also
```

Then use it in `repo/Item`:

```go
type Item interface {
  Search(context.Context, *item.Query, *usecasex.Pagination) (item.List, *usecasex.PageInfo, error)
}
```